### PR TITLE
linux-arm64 build

### PR DESCRIPTION
**Pull request by iameli**
Three big changes here:

1. Implemented cross-compilation for linux-arm64 using `clang`; this is triggered with `GOARCH=arm64` just like we do for darwin-arm64.
2. Removed the multiple-Dockerfile mess and replaced them with a single, unified multi-stage Dockerfile
3. Switched to `docker build buildx` for the Docker build process, which uses the above cross-compilation to build and publish an arm64 go-livepeer build alongside our existing amd64 build. [Info on how that works here](https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/).

~~One note: I had to upgrade from Ubuntu 18.04 to Ubuntu 20.04 because I needed clang 10 - some bug related to `fastmath` stuff when we were on clang 8. This will bump our glibc dependency by two years. Possible we should either figure out how to downpatch or look into `musl`.~~ Fixed, build now works with Ubuntu 18.04.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+COPY go.mod go.mod
+COPY go.sum go.sum
+RUN go mod download
+
+COPY . .
+ENV CPATH /usr/aarch64-linux-gnu/include
+RUN make livepeer livepeer_cli livepeer_bench livepeer_router
+
+CMD ./upload_build.sh
```
Comment (hjpotter92) [inline docker/Dockerfile]: is upload needed from inside the container?

File: install_ffmpeg.sh [hunk 2]
```diff
@@ -77,7 +87,7 @@ fi
 if [[ ! -e "$ROOT/x264" ]]; then
   git clone http://git.videolan.org/git/x264.git "$ROOT/x264"
   cd "$ROOT/x264"
-  if [[ $IS_M1 ]]; then
+  if [[ $IS_ARM64 ]]; then
     # newer git master, compiles on Apple Silicon
```
Comment (hjpotter92) [inline install_ffmpeg.sh:91]: and `arm64` ;)

File: .github/workflows/docker.yaml [hunk 3]
```diff
@@ -53,7 +41,12 @@ jobs:
           # Our Docker tag name should be our branch name with just alphanums
           BRANCH_TAG="$(echo $GHA_REF | sed -e 's:refs/heads/::; s:/:-:g' | tr -cd '[:alnum:]_-')"
           VERSION_TAG=$(./print_version.sh)
-          docker build -t current-build -f docker/Dockerfile.release-linux .
+          docker buildx \
```
Comment (hjpotter92) [inline .github/workflows/docker.yaml]: would it support multi-platform builds for docker images as well? I can update this flow accordingly then.

Comment (iameli) [reply .github/workflows/docker.yaml]: I did that once, and it worked, but it took like 75 minutes, which would be even slower than the Windows build. We might be better off using the cross-compiler for the arm64 Docker image too... can different stages of a multistage Docker build use different architectures?

File: docker/Dockerfile (hunk 1 above)
Comment (iameli) [reply docker/Dockerfile]: Not anymore, good point

File: .github/workflows/docker.yaml (hunk 3 above)
Comment (iameli) [reply .github/workflows/docker.yaml]: They can, and I implemented it, woo

**Review by leszko — COMMENTED** [review]
@iameli how about keeping the reviewers list shorter? I think 1-2 reviewers should be enough for this PR.

**Review by hjpotter92 — APPROVED** [review]

**Comment by iameli** [comment]
@leszko will do!

**Comment by iameli** [comment]
Attempting to get a build with 18.04 working for maximum compatibility, think this maybe will do it...

**Pull request merged**